clang/tools/cir-lsp-server/CMakeLists.txt

```diff
@@ -24,12 +20,25 @@ set(LIBS
   MLIRIR
   )
 
+set(LIBS
+  clangCIR
+  clangCIRLoweringThroughMLIR
+  clangCIRLoweringDirectToLLVM
+  MLIRCIR
+  )
+
+set(DEPS ${LIBS})
+if (NOT CLANG_BUILT_STANDALONE)
+  list(APPEND DEPS ${MLIR_LIBS})
+endif()
+
 add_mlir_tool(cir-lsp-server
   cir-lsp-server.cpp
 
   DEPENDS
-  ${LIBS}
+  ${DEPS}
```

Comment:
In general, I think we want to only be linking the MLIR libs with `mlir_target_link_libraries` and not under `DEPENDS`, so that we link against libMLIR.dylib correctly.